front/src/applications/editor/tools/rangeEdition/components/RangeEditionLeftPanel.tsx

```diff
@@ -77,75 +108,109 @@ const RangeEditionLeftPanel = () => {
       entity: newEntity,
     });
   };
+
   const unselectSwitch = (swId: string) => () => {
+    const filteredSelectedSwitches = Object.fromEntries(
+      Object.entries(selectedSwitches).filter(([key]) => key !== swId)
+    );
+    if (isEmptyArray(Object.keys(filteredSelectedSwitches))) resetSpeedRestrictionSelections();
     setState({
-      selectedSwitches: selectedSwitches.filter((s) => s !== swId),
+      selectedSwitches: filteredSelectedSwitches,
     });
   };
 
+  const handleRouteClicked = (select: boolean) => async (routeId: string) => {
+    if (!isEmpty(routeElements)) {
```

Comment:
i think you can use a return state if (isEmpty(routeElements)) return and then you don't have to wrap the rest

Reply:
done